### PR TITLE
Added doubly linked list in C

### PR DESCRIPTION
**Pull request by gauravburjwal**
I'm GSSoC participant
I have added Doubly Linked List in C
Refer to issue #718

### CODE REVIEW

File: Doubly_Linked_List/Doubly_Linked_List.c [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+{
+    struct node *ptr;
+    ptr = start;
+    while(ptr -> next != NULL)
+    {
+        ptr = ptr -> next;
+    }
+    ptr -> prev -> next = NULL;
+    free(ptr);
+    return start;
```
Comment (poush) [inline Doubly_Linked_List/Doubly_Linked_List.c]: These are all pointers so what's the need to return them? when the original chain will be modified. You are just returning the address you are receiving as an argument.

Comment (gauravburjwal) [reply Doubly_Linked_List/Doubly_Linked_List.c]: Yes you are right. Do I need to change it? Or it will work ? While compiling there is no error.

Comment (jainaman224) [reply Doubly_Linked_List/Doubly_Linked_List.c]: It would be great if you can change them.

Comment (gauravburjwal) [reply Doubly_Linked_List/Doubly_Linked_List.c]: As of now it is difficult for me to change. I am still learning about pointers and data structures.

**Pull request merged**